zrml/prediction-markets/src/lib.rs

```diff
@@ -135,21 +146,45 @@ mod pallet {
             origin: OriginFor<T>,
             market_id: MarketIdOf<T>,
         ) -> DispatchResultWithPostInfo {
-            T::DestroyOrigin::ensure_origin(origin)?;
+            ensure_root(origin)?;
 
             let mut total_accounts = 0usize;
             let mut share_accounts = 0usize;
             let market = T::MarketCommons::market(&market_id)?;
             let market_status = market.status;
             let outcome_assets = Self::outcome_assets(market_id, &market);
             let outcome_assets_amount = outcome_assets.len();
+
+            // Slash outstanding bonds
+            if market_status == MarketStatus::Proposed {
```

Comment:
So the AdvisoryBond needs only to get slashed in the case `no state -> Proposed for advised`

`permissionless` has `N/A` (not available?) so no state which would lead to the proposed market status.

Yeah, then this case would only get called when it is advised. But how do you ensure, that it has "no state" before? Because there is no other possibility for "advised" to get into the `proposed` state?

Reply:
I'm honestly not quite sure why (in the audit) state _transitions_ are associated with slashes. We need to slash according to _market state_ and _creation_. Depending on the state and the creation type, different bonds are reserved and now need to be slashed.

Reply:
See below for an example.